op-supervisor/supervisor/backend/source/log_processor.go

```diff
@@ -0,0 +1,55 @@
+package source
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/ethereum-optimism/optimism/op-service/eth"
+	"github.com/ethereum-optimism/optimism/op-supervisor/supervisor/backend/db"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/crypto"
+)
+
+type LogStorage interface {
+	AddLog(logHash db.TruncatedHash, block eth.BlockID, timestamp uint64, logIdx uint32, execMsg *db.ExecutingMessage) error
+}
+
+type logProcessor struct {
+	logStore LogStorage
+}
+
+func newLogProcessor(logStore LogStorage) *logProcessor {
+	return &logProcessor{logStore}
+}
+
+func (p *logProcessor) ProcessLogs(_ context.Context, block eth.L1BlockRef, rcpts types.Receipts) error {
+	for _, rcpt := range rcpts {
+		for _, l := range rcpt.Logs {
+			logHash := logToHash(l)
+			err := p.logStore.AddLog(logHash, block.ID(), block.Time, uint32(l.Index), nil)
+			if err != nil {
+				// TODO(optimism#11044): Need to roll back to the start of the block....
+				return fmt.Errorf("failed to add log %d from block %v: %w", l.Index, block.ID(), err)
+			}
+		}
+	}
+	return nil
+}
+
+func logToHash(l *types.Log) db.TruncatedHash {
+	payloadHash := crypto.Keccak256(logToPayload(l))
+	msg := make([]byte, 0, 2*common.HashLength)
+	msg = append(msg, l.Address.Bytes()...)
+	msg = append(msg, payloadHash...)
+	return db.TruncateHash(crypto.Keccak256Hash(msg))
+}
+
+func logToPayload(l *types.Log) []byte {
+	msg := make([]byte, 0)
+	for _, topic := range l.Topics {
+		msg = append(msg, topic.Bytes()...)
+	}
+	msg = append(msg, l.Data...)
+	return msg
+}
```

Comment:
Should we encode the number of topics as prefix, so we know where the trailing non-topic data starts?

Reply:
I was following https://specs.optimism.io/interop/messaging.html#message-payload

We could change that if there's a reason to, but would need to update the contracts as well so I think they're just emitting the payload hash now rather than the full payload data.  In terms of our use cases I don't think we ever decode that message payload - we just want to check it matches what was claimed by the executing message.

For other people using this, if you have the original ABI then you would know how many topics there are and would have the other info like types required to decode the whole message. According to the spec you can just use `abi.decode` though I don't know enough about `abi.decode` to know that for sure.

Reply:
ah the change to only emit the payload hash is still a spec PR (https://github.com/ethereum-optimism/specs/pull/254/).